sct-commons/src/main/java/org/lfenergy/compas/sct/commons/LNodeStatusService.java

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2024 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons;
+
+import lombok.RequiredArgsConstructor;
+import org.lfenergy.compas.scl2007b4.model.*;
+import org.lfenergy.compas.sct.commons.domain.DataAttribute;
+import org.lfenergy.compas.sct.commons.domain.DoLinkedToDa;
+import org.lfenergy.compas.sct.commons.domain.DoLinkedToDaFilter;
+import org.lfenergy.compas.sct.commons.dto.SclReportItem;
+import org.lfenergy.compas.sct.commons.util.CommonConstants;
+import org.lfenergy.compas.sct.commons.util.PrivateUtils;
+import org.lfenergy.compas.sct.commons.util.SclConstructorHelper;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@RequiredArgsConstructor
+public class LNodeStatusService {
+
+    private static final String LNODE_STATUS_PRIVATE_TYPE = "COMPAS-LNodeStatus";
+    private final LdeviceService ldeviceService;
+    private final LnService lnService;
+    private final DataTypeTemplatesService dataTypeTemplatesService;
+
+    public List<SclReportItem> updateLnModStValBasedOnLNodeStatus(SCL scl) {
+        return scl.getSubstation().stream()
+                .flatMap(tSubstation -> tSubstation.getVoltageLevel().stream())
+                .flatMap(tVoltageLevel -> tVoltageLevel.getBay().stream())
+                .flatMap(tBay -> tBay.getFunction().stream())
+                .flatMap(tFunction -> tFunction.getLNode().stream())
+                .map(tlNode -> updateSingleLnModStValBasedOnLNodeStatus(scl, tlNode))
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    private SclReportItem updateSingleLnModStValBasedOnLNodeStatus(SCL scl, TLNode tlNode) {
+        String lNodeLNS = PrivateUtils.extractStringPrivate(tlNode, LNODE_STATUS_PRIVATE_TYPE).orElse(null);
+        if (!"on".equals(lNodeLNS) && !"off".equals(lNodeLNS)) {
+            return SclReportItem.error(lNodePath(tlNode), "The private %s of the LNode has invalid value. Expecting one of [on, off] but got : %s".formatted(LNODE_STATUS_PRIVATE_TYPE, lNodeLNS));
+        }
+        TAnyLN anyLn = findLn(scl, tlNode.getIedName(), tlNode.getLdInst(), tlNode.getLnClass().getFirst(), tlNode.getLnInst(), tlNode.getPrefix()).orElse(null);
+        if (anyLn == null) {
+            return SclReportItem.error(lNodePath(tlNode), "LNode in Substation section does not have a matching LN in IED section");
+        }
+        String anyLnLNS = PrivateUtils.extractStringPrivate(anyLn, LNODE_STATUS_PRIVATE_TYPE).orElse(null);
```

Comment:
Similar to LNode, there are no checks for the Private's content when it appears in an LN object ?